cmd/kyma/apply/function/function.go

```diff
@@ -103,32 +105,45 @@ func (c *command) Run() error {
 		return err
 	}
 
-	subscriptions, err := resources.NewSubscriptions(configuration)
-	if err != nil {
-		step.Failure()
-		return err
-	}
+	operators := []operator.Operator{}
 
-	kymaAddress, err := c.kymaHostAddress()
-	if err != nil {
-		step.LogErrorf("%s\n%s", err, "Check if your cluster is available and has Kyma installed.")
+	if configuration.Subscriptions != nil {
+
+		subscriptionGVR := operator.SubscriptionGVR(configuration.SchemaVersion)
+		err := isDependencyInstalled(client, subscriptionGVR)
+		if err != nil {
+			step.Failure()
+			return err
+		}
+
+		subscriptions, err := resources.NewSubscriptions(configuration)
+		if err != nil {
+			step.Failure()
+			return err
+		}
+		operators = append(operators, operator.NewSubscriptionOperator(client.Resource(subscriptionGVR).Namespace(configuration.Namespace),
+			configuration.Name, configuration.Namespace, subscriptions...))
 	}
 
-	apiRules, err := resources.NewAPIRule(configuration, kymaAddress)
-	if err != nil {
-		step.Failure()
-		return err
+	if configuration.APIRules != nil {
```

Comment:
Same thing as for the `configuration.Subscriptions` field

```suggestion
	if len(configuration.APIRules) != 0 {
```